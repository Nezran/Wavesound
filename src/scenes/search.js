import React from 'react';
import TextField from 'material-ui/TextField';
import { connect } from 'react-redux'
import * as actions from '../actions/index'
import { bindActionCreators } from 'redux'

import CircularProgress from 'material-ui/CircularProgress';

/*const styles = {
    block: {
        maxWidth: 250,
    },
    radioButton: {
        marginBottom: 16,
        float:'left'
    },
};*/

class Search extends React.Component{

    constructor(props){
        super(props)

        console.log(this.props)

        this.handleChange = this.handleChange.bind(this);
        this.handleSelect = this.handleSelect.bind(this);

    }

    handleChange(event){

       // this.props.actions.stringSearch(event.target.value)
        //actions.stringSearch(event.target.value,'artist')
        this.props.actions.stringSearch(event.target.value)
        /*ici middleware */

    }

    handleSelect(event){

        //actions.typeSearch(event.target.value)
        this.props.actions.typeSearch(event.target.value)

        /*ici middleware */
    }

    render(){

        return(
            <div style={{width:'400px', margin:'0 auto'}}>
                <TextField
                    onChange={this.handleChange}
                    floatingLabelText={'Rechercher un artiste...'}
                    value={this.props.state.params.string}
            />{
                this.props.state.artistes.isFetching ? <CircularProgress size={60} thickness={7} /> : ''
            }
                <br/>
            </div>
        )

    }
}


const mapStateToProps = (state) => ({
    state : state,
})

const mapDispatchToProps = (dispatch) => ({
    actions : bindActionCreators(actions,dispatch)
})

Search = connect(mapStateToProps, mapDispatchToProps)(Search)


export default Search

