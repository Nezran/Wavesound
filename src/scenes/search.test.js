import React from 'react'
import search from './search'
import renderer from 'react-test-renderer';



test('Link changes the class when hovered', () => {
    const component = renderer.create(
        <search/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const componentProps = renderer.create(
        <search state={{artistes:{isFetching:true}}}/>,
    );
    let treeProps = componentProps.toJSON();
    expect(treeProps).toMatchSnapshot();






});