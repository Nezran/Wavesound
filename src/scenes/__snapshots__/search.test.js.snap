// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `<search />`;

exports[`Link changes the class when hovered 2`] = `<search />`;

exports[`Link changes the class when hovered 3`] = `
<search
  state={
    Object {
      "artistes": Object {
        "isFetching": true,
      },
    }
  }
/>
`;
